Add support for indicating if a global tracer has been registered

MikeGoldsmith opened the pull request:
Adds support for knowing if a global tracer has been registered via a global variable including tests.

NOTE: Python currently does not prevent you re-registering a global tracer multiple times like C# and Java does.

carlosalberto commented on opentracing/__init__.py:
```diff
@@ -58,3 +59,14 @@ def set_global_tracer(value):
     global tracer
     tracer = value
+
+    is_registered = True
```
You need to declare `global is_registered` for it being actually set (as done with `global tracer`). Also, I'd rename it to `is_tracer_registered` ;)

carlosalberto commented:
I'm ok with this change, but I'd like to hear what the other maintainers think ;)

MikeGoldsmith commented:
@carlosalberto I've updated the PR to make the variable global and renamed to `is_tracer_registered`. However, I see the test that verifies the `is_global_tracer_registered` returns _True_ after registering a tracer fails. Any idea what is needed to fix it?

pglombardo commented:
Just a curiosity:  Couldn't this PR be reduced to:

```
def is_global_tracer_registered():
    global tracer
    return type(tracer) is Tracer
```

carlosalberto commented:
Hey @pglombardo 

So the original PR (in Java - https://github.com/opentracing/opentracing-java/pull/321) was created to address precisely this check.

By default, the global tracer value is `Tracer` (which would mean no global tracer is installed), but we want to allow users to explicitly set it to `Tracer` anyway:

```python
opentracing.set_global_tracer(opentracing.Tracer()) # is_global_tracer_registered() returns False now
```

Hope that makes sense ;)

carlosalberto commented:
btw @MikeGoldsmith I think we are cool to merge (@yurishkuro had approved the latest changes on top of this, which I merged moments ago), but we need to merge #113 first.

Once that is done, we should be able to merge ;)

MikeGoldsmith commented:
Thank @carlosalberto. Let me know if I can help with anything to get this change merged.

carlosalberto commented:
Merged, thanks for the PR @MikeGoldsmith !
